godot-core/src/engine.rs

```diff
@@ -16,6 +16,44 @@ pub use crate::gen::central::global;
 pub use crate::gen::classes::*;
 pub use crate::gen::utilities;
 
+use self::packed_scene::GenEditState;
+
+/// Extension trait for convenience functions on `PackedScene`
+pub trait PackedSceneExt {
+    /// ⚠️ Instantiates the scene as type `T`, panicking if not found or bad type.
+    ///
+    /// # Panics
+    /// If the scene is not type `T` or inherited.
+    fn instantiate_as<T>(&self) -> Gd<T>
+    where
+        T: Inherits<Node>,
+    {
+        self.try_instantiate_as::<T>()
+            .unwrap_or_else(|| panic!("Failed to instantiate {to}", to = T::CLASS_NAME,))
+    }
+
+    /// Instantiates the scene as type `T` (fallible).
+    ///
+    /// If the scene is not type `T` or inherited.
+    fn try_instantiate_as<T>(&self) -> Option<Gd<T>>
+    where
+        T: Inherits<Node>;
+}
+
+impl PackedSceneExt for PackedScene {
+    fn try_instantiate_as<T>(&self) -> Option<Gd<T>>
+    where
+        T: Inherits<Node>,
+    {
+        self.instantiate(GenEditState::GEN_EDIT_STATE_DISABLED)
+            .map(|gd| {
+                gd.try_cast::<T>().unwrap_or_else(|| {
+                    panic!("Failed to cast {self:?} into {to}", to = T::CLASS_NAME,)
+                })
+            })
```

Comment:
You still panic here. The idea of `try_` functions is to _not_ panic, but propagate the error (or in this case, `None`).